src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataPropertyFilter.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class SaveTempDataPropertyFilter : ISaveTempDataCallback
+    {
+        public string Prefix { get; set; }
+
+        public object Subject { get; set; }
+
+        public IDictionary<PropertyInfo, object> OriginalValues { get; set; }
+
+        public void OnTempDataSaving(ITempDataDictionary tempData)
+        {
+            if (Subject != null && OriginalValues != null)
+            {
+                foreach (var kvp in OriginalValues)
+                {
+                    var property = kvp.Key;
+                    var originalValue = kvp.Value;
+
+                    var newValue = property.GetValue(Subject);
+                    if (newValue != null && newValue != originalValue)
```

Comment:
@danroth27 had a question about what we'd need to preserve this across multiple requests. Resetting the value wouldn't work, would it?